examples/timebase_master.py

```diff
@@ -0,0 +1,148 @@
+        else:
+            # Compute BBT info based on previous period.
+            pos.tick += int(nframes * pos.ticks_per_beat *
+                            pos.beats_per_minute / (pos.frame_rate * 60))
+
+            while pos.tick >= pos.ticks_per_beat:
+                pos.tick -= int(pos.ticks_per_beat)
+                pos.beat += 1
+
+                if pos.beat > pos.beats_per_bar:
+                    pos.beat = 1
+                    pos.bar += 1
+                    pos.bar_start_tick += pos.beats_per_bar * pos.ticks_per_beat
+
+                    if self.debug:
+                        print("Pos:", jack.position2dict(pos))
+
+    def become_timebase_master(self, conditional=None):
+        return self.set_timebase_callback(self._tb_callback, conditional
+                                          if conditional is not None
+                                          else self.conditional)
+
+
+def main(args=None):
+    ap = argparse.ArgumentParser(description=__doc__)
+    ap.add_argument(
+        '-d', '--debug',
+        action='store_true',
+        help="Enable debug messages")
+    ap.add_argument(
+        '-c', '--conditional',
+        action='store_true',
+        help="Exit if another timebase master is already active")
+    ap.add_argument(
+        '-n', '--client-name',
+        metavar='NAME',
+        default='timebase',
+        help="JACK client name (default: %(default)s)")
+    ap.add_argument(
+        '-m', '--meter',
+        default='4/4',
+        help="Meter as <beats-per-bar>/<beat-type> (default: %(default)s)")
+    ap.add_argument(
+        '-t', '--ticks-per-beat',
+        type=int,
+        metavar='NUM',
+        default=1920,
+        help="Ticks per beat (default: %(default)s)")
+    ap.add_argument(
+        'tempo',
+        nargs='?',
+        type=float,
+        default=120.0,
+        help="Tempo in beats per minute (default: %(default)s)")
+
+    args = ap.parse_args(args if args is not None else sys.argv[1:])
+
+    try:
+        beats_per_bar, beat_type = (int(x) for x in args.meter.split('/', 1))
+    except (TypeError, ValueError):
+        print("Error: invalid meter: %s\n" % args.meter)
+        ap.print_help()
+        return 2
+
+    try:
+        tbmaster = TimebaseMasterClient(
+            args.client_name,
+            bpm=max(0.0, min(300.0, args.tempo)),
+            beats_per_bar=beats_per_bar,
+            beat_type=beat_type,
+            ticks_per_beat=args.ticks_per_beat,
+            debug=args.debug)
+    except jack.JackError as exc:
+        return "Could not create timebase master JACK client: %s" % exc
+
+    with tbmaster:
+        if tbmaster.become_timebase_master(args.conditional):
+            try:
+                print("Press Ctrl-C to quit...")
+                tbmaster.stop_event.wait()
+            except KeyboardInterrupt:
+                print('')
+            finally:
+                try:
+                    tbmaster.release_timebase()
```

Comment:
There seems to be an inconsistency:

* `client.set_sync_callback(cb)` vs. `client.set_sync_callback(None)`
* `client.set_timebase_callback(cb)` vs. `client.release_timebase()`

Or am I missing something?

This is of course directly taken from JACK, but should we fix this as long as `release_timebase()` is still very fresh?

But I guess the problem is the second argument, right?

Reply:
I don't think we should change names taken from libjack without a good reason (except the removal of the `jack_` prefix). It makes it harder to use the libjack API docs.

We could of course, in addition to having `release_timebase`, augment `set_timebase_callback` to call `release_timebase` if called with a `None` argument, but that would have to be another PR and this script wouldn't need to change.